services/motion/builtin/move_request.go

```diff
@@ -653,9 +665,23 @@ func (ms *builtIn) relativeMoveRequestFromAbsolute(
 		return nil, err
 	}
 	startPoseInv := spatialmath.PoseInverse(startPose.Pose())
-
 	goal := referenceframe.NewPoseInFrame(referenceframe.World, spatialmath.PoseBetween(startPose.Pose(), goalPoseInWorld))
 
+	// Here we determine if we already are at the goal
+	// If our motion profile is position_only then, we only check against our current & desired position
+	// Conversely if our motion profile is anything else, then we also need to check again our
+	// current & desired orientation
+	if valExtra.motionProfile != motionplan.PositionOnlyMotionProfile {
```

Comment:
@biotinker I think this ought to do the trick.

Reply:
You've got your `if` flipped here; right now you're checking orientation if the profile *is* position-only, and skipping the orientation check if not.

Otherwise yes this should be fine.

Reply:
oops!